src/core/annotation.js

```diff
@@ -746,6 +739,37 @@ var LinkAnnotation = (function LinkAnnotationClosure() {
   return LinkAnnotation;
 })();
 
+var PopupAnnotation = (function PopupAnnotationClosure() {
+  function PopupAnnotation(parameters) {
+    Annotation.call(this, parameters);
+
+    this.data.annotationType = AnnotationType.POPUP;
+
+    var dict = parameters.dict;
+    if (!dict.has('Parent')) {
+      warn('Popup annotations without a parent annotation are not supported.');
+      return;
+    }
+
+    var parentItem = dict.get('Parent');
+    if (!parentItem) {
+      warn('Popup annotation has an invalid parent annotation.');
+      return;
+    }
+    this.data.parentId = dict.map.Parent.num;
```

Comment:
The main problem with using `Dict.get('Parent')` is that the ID (`num`) is not in the returned object anymore. Do you know of a better way of obtaining this ID other than the method I used?